src/main/java/net/shadowmage/ancientwarfare/npc/item/ItemCommandBaton.java

```diff
@@ -59,28 +60,22 @@ public ItemCommandBaton(String name, ToolMaterial material) {
 	@Override
 	@SideOnly(Side.CLIENT)
 	public void addInformation(ItemStack stack, @Nullable World worldIn, List<String> tooltip, ITooltipFlag flagIn) {
+		if (stack.getTagCompound() == null){
+			stack.setTagInfo("baton_mode", stack.serializeNBT());
+			stack.getTagCompound().setByte("baton_mode", (byte) BatonMode.CLEAR_COMMAND.ordinal());
+		}
+
 		String keyText, text;
 		text = "RMB" + " = " + I18n.format("guistrings.npc.baton.add_remove");
 		tooltip.add(text);
 
 		keyText = InputHandler.ALT_ITEM_USE_1.getDisplayName();
-		text = keyText + " = " + I18n.format("guistrings.npc.baton.clear");
+		BatonMode mode = BatonMode.values()[stack.getTagCompound().getByte("baton_mode")];

```

Comment:
I am not fan of using indexes of enums in nbt for several reasons:
1. if you look at the nbt data you're not clear what mode is set
2. if we move the element in the enum that should have no effect on data saved in nbt
3. if you actually use that index directly to get enum instance at the index of enum's values() that will break the game if the index for some reason gets corrupted and returns value out of range

I would much rather like to see a name used here similar to what's done for example with storing variant of wooden coffin (BlockWoodenCoffin.Variant .getName() / .fromName() )